### ISSUE TITLE
[SR-9094] Unavailable enum case with synthesized raw value causes error

### ISSUE THREAD

**Issue by swift-ci** [apple/swift#51591]
|                  |                 |
   |------------------|-----------------|
   |Previous ID       | SR-9094      |
   |Radar             | None         |
   |Original Reporter | owenvoorhees (JIRA User)      |
   |Type              | Bug    |
   |Status            | In Progress        |
   |Resolution        |     |
        
   <details>
  <summary>Additional Detail from JIRA</summary>

   |                  |                 |
   |------------------|-----------------|
   |Votes             | 0         |
   |Component/s       | Compiler    |
   |Labels            | Bug        |
   |Assignee          | None      |
   |Priority          | Medium      |

   

   md5: 3956e1b34a5c5b310af7f0279053078c

  </details>

**relates to**:  
* [SR-7151](https://bugs.swift.org/browse/SR-7151) Generalize CaseIterable Synthesis for Availability  
* [SR-8634](https://bugs.swift.org/browse/SR-8634) Deprecation warning at declaration site in typed enum  

**Issue Description:**

The following code compiles as expected:

```swift
enum TrafficLight {
    @available(*, unavailable) case red
    case green
    case yellow
}
```

But this code leads to a compiler error:

```swift
enum TrafficLight: String {
    @available(*, unavailable) case red
    case green
    case yellow
}
```
 
```
MyPlayground.playground:4:37: note: 'red' has been explicitly marked unavailable here  
@available(\*, unavailable) case red  
~~~~~~~~~~~~~~~~~~~~~~~~~ ^
```
 
Some initial investigation indicates the error is may be related to the synthesized `RawRepresentable` conformance tripping availability checking

**Comment by lordzsolt** [comment]
It looks like this got solved at one point, the code now compiles correctly.

**Comment by AnthonyLatsis** [comment]
Pretty sure this was fixed by #22198.
